### PR TITLE
[plugin-web-app] Refactor window steps to allow wait for window

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/SetContextSteps.java [hunk 1]
```diff
@@ -291,58 +297,50 @@ public void switchingToWindow()
...
+    * <li>Searches among currently opened windows (and tabs) for a window with the specified <b>windowName</b>.</li>
+    * <li>If such window is found switches focus to it. If window is not found current focus stays unchanged;</li>
+    * </ul>
+    * @param comparisonRule is equal to, contains, does not contain
+    * @param title Value of the {@code <title>} tag of a desired window
+    * @param duration in format <a href="https://en.wikipedia.org/wiki/ISO_8601">Duration Format</a>
+    * @see <a href="https://html.spec.whatwg.org/#browsing-context"><i>Browsing context (Window &amp; Document)</i></a>
+    * @see <a href="https://www.w3schools.com/tags/default.asp"><i>HTML Element Reference</i></a>
+    */
+    @When("I switch to window where title $comparisonRule `$title` waiting `$duration`")
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/SetContextSteps.java]: I think there should be 2 types of steps: waiting for window and switching to window

File: vividus-bdd-engine/src/test/java/org/vividus/bdd/steps/StringComparisonRuleTests.java [hunk 2]
```diff
@@ -36,7 +37,8 @@
         return Stream.of(
             Arguments.of(StringComparisonRule.IS_EQUAL_TO,      IsEqual.class),
             Arguments.of(StringComparisonRule.CONTAINS,         StringContains.class),
-            Arguments.of(StringComparisonRule.DOES_NOT_CONTAIN, IsNot.class)
+            Arguments.of(StringComparisonRule.DOES_NOT_CONTAIN, IsNot.class),
+            Arguments.of(StringComparisonRule.MATCHES, StringRegularExpression.class)
```
Comment (valfirst) [inline vividus-bdd-engine/src/test/java/org/vividus/bdd/steps/StringComparisonRuleTests.java]: keep aligning as table

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/SetContextSteps.java [hunk 3]
```diff
@@ -291,58 +298,84 @@ public void switchingToWindow()
...
+    * @param duration in format <a href="https://en.wikipedia.org/wiki/ISO_8601">Duration Format</a>
+    * @see <a href="https://html.spec.whatwg.org/#browsing-context"><i>Browsing context (Window &amp; Document)</i></a>
+    * @see <a href="https://www.w3schools.com/tags/default.asp"><i>HTML Element Reference</i></a>
+    */
+    @When("I wait `$duration` until window with title that $comparisonRule `$title` appears and switch to it")
+    public void waitForWindowAndSwitch(Duration duration, StringComparisonRule comparisonRule, String title)
     {
-        switchToWindow(containsString(windowPartName));
+        Matcher<String> expected = comparisonRule.createMatcher(title);
+        WaitResult<Boolean> result = waitActions.wait(webDriverProvider.get(), duration, new ExpectedCondition<>()
```
Comment (uarlouski) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/SetContextSteps.java]: does it makes sense to move out expected condition to separate class as we usually do it?

**Comment by valfirst** [comment]
The changes have been cherry-picked to `master` branch

**Pull request closed without merging**